addons/mail/static/src/new/core/thread_service.js

```diff
@@ -475,7 +474,10 @@ export class ThreadService {
                 "message_needaction_counter",
                 "name",
                 "state",
+                "group_based_subscription",
             ]);
+            thread.channel = thread.channel ?? {};
```

Comment:
I feel there should be conditions on it, and it shouldn't be defined at all for non channel threads.